File: payment-api-node/controllers/checkout.controller.js
const Order = require("../models/order.model");

exports.createOrder = async (req, res) => {
  try {
    const { fullName, address, phone, paymentMethod, items } = req.body;

    if (!fullName || !address || !phone || !paymentMethod || !items || items.length === 0) {
      return res.status(400).json({ success: false, message: "Thiếu thông tin đơn hàng" });
    }

    const totalAmount = items.reduce((sum, item) => sum + item.quantity * item.unitPrice, 0);
    const userId = req.user?.id || "anonymous"; // nếu có xác thực

    const order = new Order({
      userId,
      fullName,
      address,
      phone,
      paymentMethod,
      totalAmount,
      items
    });

    await order.save();
    res.status(201).json({ success: true, orderId: order._id });
  } catch (err) {
    console.error("Lỗi khi tạo đơn hàng:", err.message);
    res.status(500).json({ success: false, message: "Lỗi máy chủ" });
  }
};

File: payment-api/Controllers/CheckoutController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using PaymentApi.Data;
using PaymentApi.Models;
using System.Security.Claims;

namespace PaymentApi.Controllers;

[ApiController]
[Route("api/[controller]")]
public class CheckoutController : ControllerBase
{
    private readonly ApplicationDbContext _db;

    public CheckoutController(ApplicationDbContext db)
    {
        _db = db;
    }

    [HttpPost]
    [Authorize] // Nếu hệ thống đã có xác thực
    public IActionResult Post([FromBody] CheckoutRequest request)
    {
        if (!ModelState.IsValid || request.Items == null || !request.Items.Any())
            return BadRequest(new { success = false, message = "Dữ liệu không hợp lệ" });

        var userId = User.FindFirst(ClaimTypes.NameIdentifier)?.Value ?? "anonymous";

        var order = new Order
        {
            UserId = userId,
            FullName = request.FullName,
            Address = request.Address,
            Phone = request.Phone,
            PaymentMethod = request.PaymentMethod,
            Items = request.Items,
            TotalAmount = request.Items.Sum(i => i.Quantity * i.UnitPrice),
            Status = OrderStatus.Pending
        };

        _db.Orders.Add(order);
        _db.SaveChanges();

        return Ok(new { success = true, orderId = order.Id });
    }
}
